ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -792,7 +795,22 @@ void initialize(int index, CompositeBlobInfo.ChunkMetadata chunkMetadata) {
           RouterOperation.GetBlobOperation, chunkBlobId);
       progressTracker = new ProgressTracker(chunkOperationTracker);
       state = ChunkState.Ready;
-      initializedTimeMs = System.currentTimeMillis();
+      initializedTimeMs = time.milliseconds();
+    }
+
+    /**
+     * Reset some variables for retry.
+     */
+    void resetForRetry() {
+      logger.trace("BlobId {}: Retry for chunk Id: {}, failed attempts: {}, initializedTime: {}", blobId, chunkBlobId,
+          failedAttempts, initializedTimeMs);
+      chunkException = null;
+      chunkOperationTracker = getOperationTracker(chunkBlobId.getPartition(), chunkBlobId.getDatacenterId(),
+          RouterOperation.GetBlobOperation, chunkBlobId);
+      progressTracker = new ProgressTracker(chunkOperationTracker);
+      state = ChunkState.Ready;
+      failedAttempts++;
+      routerMetrics.getBlobRetryCount.inc();
```

Comment:
If for some reason, the servers returns the server unavailable very quickly and there are not many replicas online for this partition, our frontend retry may be too stressful to these remaining replicas.

I was thinking about some simple backoff.
But we cannot do 
"sleep(10ms * failedAttempts);"
It'll block the RequestResponseHandlerThread which is shared by different Operations.

Do we have some other way for some simple backoff? Probably record the time and when it reaches that time + delay time, start the next retry. Not sure if the code will be too complicated.

Reply:
kinda hard to do this, we have to add another configuration for backoff, the time to sleep etc... we will be introducing too many new configurations.